src/components/Comment.jsx

```diff
@@ -1,24 +1,81 @@
-import { useContext } from "react";
+import { useContext, useState, useEffect } from "react";
 import { UserContext } from "../contexts/UserContext";
+import { deleteComment } from "../utils/apiRequests";
 
-const Comment = ({ comment }) => {
+const Comment = ({ comment, setComments }) => {
+  const [isUser, setIsUser] = useState(true);
   const userValue = useContext(UserContext);
   const posted = comment.created_at
     ? comment.created_at.split(/[-T:.]/)
     : ["just now."];
+
+  useEffect(() => {
+    if (comment.author === userValue.loggedInUser.username) {
+      setIsUser(true);
+    } else {
+      setIsUser(false);
+    }
+  }, [userValue.loggedInUser.username, comment.author]);
```

Comment:
You probably don't need a useEffect or state for this. You can work out `isUser` with the condition you use in your useEffect so can do it inline, it's not really a sideEffect

```js
const isUser = comment.author === userValue.loggedInUser.username
```

This way you don't need to re-set state and it's worked out correctly everytime